packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -455,10 +455,31 @@ class FlutterPlugin implements Plugin<Project> {
         return version2
     }
 
+    /**
+     * Resolves SDK version to API level. 
+     *
+     * Handles sdkVersion being integers and strings (codenames) (when the platform stability is not yet declared).
+     * Must be updated when new android SDK level is released.
+     */
+    private int resolveSdkVersion(String sdkVersion) {
+        int resolvedSdkVersion;
+        if (sdkVersion.startsWith("android-")) {
+            sdkVersion = sdkVersion.substring(8)
+        }
+
+        if (sdkVersion.isInteger()) {
+            resolvedSdkVersion = sdkVersion.toInteger()
+        } else if (sdkVersion.equals("Tiramisu")) {
+            resolvedSdkVersion = 33
+        }
```

Comment:
do you want to hard-code it? Is there a better way?

Reply:
I'm afraid there isn't. But this would have to be done only when Google releases a new Android preview, so once a year.

Still, it's simply a draft.

Reply:
cc @GaryQian - Tiramisu is still a preview, and while it's reasonable to assume Tiramisu=33, there's no guarantee that it will actually be 33.  Since strings are only used to refer to preview versions, it's fine to detect if the type is a string and exit.

Reply:
Sure, that makes sense, but it prevents developers from testing their apps on preview versions of Android. Maybe it's possible to add some escape hatch to make it possible to build Flutter apps for preview Android versions?

Reply:
yes - detectLowCompileSdkVersionOrNdkVersion can exit in line 461 if `project.android.compileSdkVersion` isn't a int.

Reply:
> `detectLowCompileSdkVersionOrNdkVersion` can exit in line 461

by "exit" you mean just assume that string is fine and not do anything, right?